src/App.jsx

```diff
@@ -0,0 +1,141 @@
+import { NavLink, Routes, Route, Link } from "react-router-dom";
+import { createContext, useState, useEffect } from "react";
+
+// components
+import Avatar from "./components/Avatar";
+import PostFeed from "./components/PostFeed";
+import Profile from "./components/Profile";
+import ClickedPost from "./components/ClickedPost";
+
+// icons
+import logoIcon from "./assets/icons/header-logo.svg";
+import profileIcon from "./assets/icons/profile.svg";
+import homeIcon from "./assets/icons/home.svg";
+import light from "./assets/icons/light.svg";
+import dark from "./assets/icons/dark.svg";
+
+// to get the active link in the nav bar
+function getClassName({ isActive }) {
+  if (isActive) {
+    return "active"; // => CSS calss
+  }
+}
+
+export const DataContext = createContext();
+
+export default function App() {
+  const [posts, setPosts] = useState([]);
+  const [contacts, setContacts] = useState([]);
+  const [user, setUser] = useState();
+  const [isLoading, setIsLoading] = useState(true);
+  const [isDark, setIsDark] = useState(getItem);
+
+  // change CSS color-scheme class based on the selected mode
+  const mode = isDark ? "dark" : "light";
+  // change the icon on the header based on the selected mode
+  const modeIcon = isDark ? dark : light;
+
+  useEffect(() => {
+    localStorage.setItem("isDark", isDark);
+  }, [isDark]);
+
+  // the fn to get the item only once (when the app starts) not every time it re-renders
+  function getItem() {
+    return localStorage.getItem("isDark") === "true";
+  }
+
+  useEffect(() => {
+    fetch("https://boolean-api-server.fly.dev/Hamada-AB/contact")
```

Comment:
1. Extract the API url into a constant in its own file that you export and import when needed, e.g.

```js
export const API_URL = "https://boolean-uk-api-server.fly.dev/"
```

Reply:
2. The API url is wrong. It's actually https://boolean-uk-api-server.fly.dev/